packages/core/src/browser/keybinding.ts

```diff
@@ -245,6 +245,22 @@ export class KeybindingRegistry {
         }
     }
 
+    /**
+     * Ensures that keybindings are inserted in order of increasing length of binding to ensure that if a
+     * user triggers a short keybinding (e.g. ctrl+k), the UI won't wait for a longer one (e.g. ctrl+k enter)
+     */
+    protected insertBindingIntoScope(item: common.Keybinding & { scope: KeybindingScope; }, scope: KeybindingScope): void {
+        const scopedKeymap = this.keymaps[scope];
+        const getNumberOfKeystrokes = (binding: common.Keybinding): number => (binding.keybinding.trim().match(/\s/)?.length ?? 0) + 1;
+        const numberOfKeystrokesInBinding = getNumberOfKeystrokes(item);
+        const indexOfFirstItemWithEqualStrokes = scopedKeymap.findIndex(existingBinding => getNumberOfKeystrokes(existingBinding) === numberOfKeystrokesInBinding);
+        if (indexOfFirstItemWithEqualStrokes > -1) {
+            scopedKeymap.splice(indexOfFirstItemWithEqualStrokes, 0, item);
+        } else {
+            scopedKeymap.push(item);
+        }
+    }
+
```

Comment:
This addition is necessary to ensure that this test passes:  
https://github.com/eclipse-theia/theia/blob/8ea255a4021393eba96a0da0acae3b1aa8f98ad6/examples/api-tests/src/keybindings.spec.js#L48-L63

Previously, it passed by coincidence - the relevant competition to the clear terminal command happened to be registered in an order that allowed the test to pass. 

FWIW, the assertion made in [this test](https://github.com/eclipse-theia/theia/blob/8ea255a4021393eba96a0da0acae3b1aa8f98ad6/examples/api-tests/src/keybindings.spec.js#L96-L115) that priority should depend on insertion order is pretty fragile, and seems designed to cover edge cases from the [`terminal` package](https://github.com/eclipse-theia/theia/blob/8ea255a4021393eba96a0da0acae3b1aa8f98ad6/packages/terminal/src/browser/terminal-frontend-contribution.ts#L430-L505) where we register a bunch of non-commands for <kbd>ctrl+\<key\></kbd> and <kbd>alt+\<key\></kbd> to get around bindings that already exist in the packages we use for terminals, but then *also* register some commands for specific combinations that we want to take priority over the dummy commands. There's got to be a better way to handle those requirements.